### PR TITLE
 Added oAuth2 type authontification to webapp v0.2

### PR DESCRIPTION
**Pull request by TaykYoku**
Added OIDC authontification to web portal.

*Core/App.py
FIX: TypeError: Expected bytes, unicode, or None; got <type 'long'> in use secret cookies

*Lib/Compiler.py
FIX: Error: Duplicate matches for "in" - could be: input-file input-js-version

*Lib/Conf.py
FIX: Added getCSSections, getCSOptions, getCSOptionsDict to help parse web.cfg

*Lib/WebHandler.py
FIX: __processCredentials: Added oidc authontificate

*WebApp/handler/AuthontificationHandler.py
NEW: Handler for auth menu

*WebApp/static/core/js/views/desktop/TaskBar.js
FIX: Added auth menu

*WebApp/static/core/js/views/tabs/SettingsPanel.js
FIX: Added auth menu

*WebApp/static/oidc/oidc-client/
NEW: Added oidc js lib

*WebApp/template/root.tpl
FIX: Added oidc js lib

*WebApp/web.cfg
FIX: Added example of authentification configuration

### CODE REVIEW

File: WebApp/handler/AuthentificationHandler.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+from WebAppDIRAC.Lib.WebHandler import WebHandler, asyncGen
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import getUsernameForID
+from DIRAC.FrameworkSystem.Client.NotificationClient   import NotificationClient
+from WebAppDIRAC.Lib import Conf
+
+import json
+import tornado
+import requests
+
+class AuthentificationHandler(WebHandler):
```
Comment (atsareg) [inline WebApp/handler/AuthentificationHandler.py]: Authentification -> Authentication

**Comment by zmathe** [comment]
@atsareg It is okay. It is better to use it from an external source. However, we can also serve this files using our web server: /opt/dirac/pro/webRoot/www/defaults/ Leave it as it is now

File: WebApp/handler/AuthentificationHandler.py (hunk 1 above)
Comment (TaykYoku) [reply WebApp/handler/AuthentificationHandler.py]: done

**Comment by atsareg** [comment]
Am I understanding correctly that this replaces #303 ?

**Pull request merged**